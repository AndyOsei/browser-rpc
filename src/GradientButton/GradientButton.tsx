import React from 'react';
import PropTypes from 'prop-types';
import Button, { ButtonProps } from '@material-ui/core/Button';

import useStyles from './styles';

export interface GradientButtonProps extends ButtonProps {
  inclination?: '45deg' | '90deg' | '180deg';
}

const GradientButton: React.FC<GradientButtonProps> = (props) => {
  const classes = useStyles(props);
  const { children, ...restProps } = props;

  return (
    <Button className={classes.root} {...restProps}>
      {children}
    </Button>
  );
};

GradientButton.defaultProps = {
  inclination: '45deg',
};

GradientButton.propTypes = {
  inclination: PropTypes.oneOf([
    '45deg',
    '90deg',
    '180deg',
  ]),
};

export default GradientButton;
