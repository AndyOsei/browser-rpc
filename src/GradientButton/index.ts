export { default } from './GradientButton';
export * from './GradientButton';
