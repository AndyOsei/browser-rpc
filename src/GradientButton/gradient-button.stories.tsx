import React from 'react';

import GradientButton from './index';

export default {
  title: 'GradientButton',
  component: GradientButton,
};

const Template = (args) => <GradientButton {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: 'Text',
};
