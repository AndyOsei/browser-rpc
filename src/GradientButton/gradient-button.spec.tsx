import React from 'react';
import renderer, {
  ReactTestRenderer,
  ReactTestInstance,
} from 'react-test-renderer';

import GradientButton from './GradientButton';

describe('GradientButton', () => {
  let component: ReactTestRenderer;
  let testInstance: ReactTestInstance;
  const text = 'Some Text';

  beforeEach(() => {
    component = renderer.create(
      <GradientButton>
        {text}
      </GradientButton>,
    );

    testInstance = component.root;
  });

  it('should display text string inside button', () => {
    expect(testInstance.findByType(GradientButton).props.children).toBe(text);
  });

  it('should add default value for inclination prop', () => {
    expect(testInstance.findByType(GradientButton).props.inclination)
      .toBe('45deg');
  });

  it('should match stapshot', () => {
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
