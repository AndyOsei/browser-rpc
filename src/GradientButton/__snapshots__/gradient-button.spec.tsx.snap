// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradientButton should match stapshot 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-root-1 makeStyles-root-4"
  disabled={false}
  inclination="45deg"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Some Text
  </span>
</button>
`;
